src/plugins/vis_type_table/public/components/table_vis_component.tsx

```diff
@@ -34,49 +63,70 @@ export const TableVisComponent = ({ table, visConfig, handlers }: TableVisCompon
       // then the row index must be adjusted as `data` has already been pruned to the page size
       adjustedRowIndex = rowIndex - pagination!.pageIndex * pagination!.pageSize;
 
-      return rows.hasOwnProperty(adjustedRowIndex)
-        ? rows[adjustedRowIndex][columnId] || null
+      return sortedRows.hasOwnProperty(adjustedRowIndex)
+        ? sortedRows[adjustedRowIndex][columnId] || null
         : null;
     }) as EuiDataGridProps['renderCellValue'];
-  }, [rows, pagination]);
-
-  // resize column
-  const [columnsWidth, setColumnsWidth] = useState<ColumnWidth[]>(
-    handlers.uiState.get('vis.columnsWidth') || []
+  }, [sortedRows, pagination]);
+
+  const dataGridColumns = getDataGridColumns(sortedRows, columns, table, handlers, columnsWidth);
+
+  const sortedColumns = useMemo(() => {
+    const sort = handlers.uiState?.get('vis.sortColumn');
+    return sort && sort.colIndex !== null && sort.direction
+      ? [{ id: dataGridColumns[sort.colIndex]?.id, direction: sort.direction }]
```

Comment:
If we eventually want `sortedColumns` by `id`, why  does the `SortColumn` type use the column index? I see lots of mappings back and forth between index and id, but it's unclear if the index value matters all that much - maybe just for the column width/resizing?

Reply:
Yeah I am aware of this. The reason why I am using index not id is that when we add/remove a column, the id might change. Below is an example that I add another column day_of_week. Count id is changed.

<img width="430" alt="Screen Shot 2022-10-10 at 19 32 13" src="https://user-images.githubusercontent.com/79961084/195156334-0a7164a1-8eeb-483b-8c55-b50d90dfe550.png">

<img width="437" alt="Screen Shot 2022-10-10 at 19 32 59" src="https://user-images.githubusercontent.com/79961084/195156339-01fdfd6f-282b-4e7e-ba3c-bf8eec95bf5a.png">

I think there might be ways to work around. but for now, using index is the safest way to guarantee we could find the correct col. 

I opened an issue to research on this to see if we could simply the code:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2549

I will work on it next sprint.